packages/x-components/src/x-modules/search/components/redirection.vue

```diff
@@ -20,103 +19,125 @@
    *
    * @public
    */
-  @Component({
-    mixins: [xComponentMixin(searchXModule)]
-  })
-  export default class Redirection extends Vue {
-    @State('search', 'redirections')
-    public redirections!: RedirectionModel[];
-
-    /**
-     * The redirection mode. Auto for auto redirection and manual for an user interaction.
-     *
-     * @public
-     */
-    @Prop({ default: 'auto' })
-    public mode!: 'auto' | 'manual';
-
-    /**
-     * The waiting time in seconds until the redirection is made.
-     *
-     * @remarks this delay only works in auto mode.
-     *
-     * @public
-     */
-    @Prop({ default: 0 })
-    public delayInSeconds!: number;
-
-    /**
-     * The timeout id, used to cancel the redirection.
-     *
-     * @internal
-     */
-    protected timeoutId?: number;
-
-    /**
-     * Boolean flag which indicates if the redirection is running.
-     *
-     * @public
-     */
-    protected isRedirecting = true;
-
-    /**
-     * Computed property which returns the first recommendation of the state, if any returns null.
-     *
-     * @returns The first redirection of the state.
-     *
-     * @internal
-     */
-    protected get redirection(): RedirectionModel | null {
-      return this.redirections?.[0] ?? null;
-    }
-
-    /**
-     * Watcher function which adds a setTimeout to the redirect method is the component
-     * is in auto mode and there are redirections.
-     *
-     * @internal
-     */
-    @Watch('redirections', { immediate: true })
-    protected redirectWithDelay(): void {
-      this.isRedirecting = true;
-      if (this.mode === 'auto' && this.redirection) {
-        // eslint-disable-next-line @typescript-eslint/unbound-method
-        this.timeoutId = window.setTimeout(this.redirect, this.delayInSeconds * 1000);
+  export default defineComponent({
+    name: 'Redirection',
+    xModule: searchXModule.name,
+    props: {
+      /**
+       * The redirection mode. Auto for auto redirection and manual for an user interaction.
+       *
+       * @public
+       */
+      mode: {
+        type: String as PropType<'auto' | 'manual'>,
+        default: 'auto'
+      },
+      /**
+       * The waiting time in seconds until the redirection is made.
+       *
+       * @remarks this delay only works in auto mode.
+       *
+       * @public
+       */
+      delayInSeconds: {
+        type: Number,
+        default: 0
       }
-    }
+    },
+    setup(props, { slots }) {
+      const $x = use$x();
```

Comment:
Could you reorder the code into blocks of const, refs, watchers, and functions? With these patterns, we improve the developer experience as we can go into any component knowing its distribution beforehand.